llmops/common/prompt_eval.py

```diff
@@ -318,34 +314,28 @@ def prepare_and_execute(
                 logger.info(df_result.head(10))
 
         if evaluator_executed and report_dir:
+            logger.info(f"Generating reports for {evaluator.name}")
             if not os.path.exists(report_dir):
                 os.makedirs(report_dir)
 
             combined_results_df = pd.concat(dataframes, ignore_index=True)
             combined_metrics_df = pd.DataFrame(metrics)
-            combined_results_df["flow_name"] = flow_name
-            combined_metrics_df["flow_name"] = flow_name
+            combined_results_df["flow_name"] = evaluator.name
+            combined_metrics_df["flow_name"] = evaluator.name
             combined_results_df["exp_run"] = flow_run
             combined_metrics_df["exp_run"] = flow_run
 
-            combined_results_df.to_csv(
-                f"{report_dir}/{run_dataset.name}_result.csv"
-                )
-            combined_metrics_df.to_csv(
-                f"{report_dir}/{run_dataset.name}_metrics.csv"
-                )
+            fname_base = f"{report_dir}/{evaluator.name}"
```

Comment:
This is the fix. `evaluator.name` is used for the file name instead of `run_dataset.name`.

This is because `run_dataset` is only a temporary variable that is used in a loop some lines above to iterate through all existing datasets. Hence, `run_dataset` here always has the same value since it's after the loop has run and has iterated through the whole list.